File: frontend/src/types/educacion.ts
export interface LeccionContenido {
  hook: string
  concept: string
  action: string
  tip: string
}

export interface LeccionData {
  id: string
  titulo: string
  titulo_en?: string
  descripcion_corta: string
  descripcion_corta_en?: string
  contenido_json: LeccionContenido
  contenido_json_en?: LeccionContenido
  nivel: 'fundamentos' | 'control' | 'crecimiento'
  duracion_minutos: number
  orden: number
  completada: boolean
}

File: backend/app/schemas/educacion.py
from pydantic import BaseModel
from typing import Optional, Any


class LeccionResponse(BaseModel):
    id: str
    titulo: str
    descripcion_corta: str
    contenido_json: dict[str, Any]
    nivel: str
    duracion_minutos: int
    orden: int
    completada: bool = False

    model_config = {"from_attributes": True}


class MarcarCompletadaResponse(BaseModel):
    leccion_id: str
    completada: bool
